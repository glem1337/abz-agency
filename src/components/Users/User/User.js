import React from 'react';
import ReactTooltip from "react-tooltip";
import defaultAvatar from '../../../assets/icons/photo-cover.svg';

const User = ({user}) => (
    <div className="user">
        <img
            className="user__photo"
            src={user.photo || defaultAvatar}
            alt={user.name}
            onError={e => {
                e.target.src = defaultAvatar;
            }}
        />
        <div className="user-info-wrapper">
            <h3 className="user__name h2">{user.name}</h3>
            <p className="user__position">{user.position}</p>
            <p className="user__email" data-tip={user.email}>{user.email}</p>
            <ReactTooltip border={false} place={'bottom'} className={'user__tooltip'}/>
            <p className="user__phone">{
                user.phone.replace(/(\d{3})(\d{2})(\d{3})(\d{2})(\d{2})/, '$1 $2 $3 $4 $5')}
            </p>
        </div>
    </div>

);

export default User;
