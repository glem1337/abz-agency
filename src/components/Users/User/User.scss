.user {
  padding-bottom: rem(38);
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  width: rem(220);
  margin: 0 auto;

  @include for-size(tablet) {
    width: rem(200);
    padding-bottom: rem(56);
  }

  @include for-size(desktop) {
    padding-bottom: rem(51);
  }

  &__photo {
    border: 2px solid $light-color;
    margin: 0 auto rem(15);
    width: rem(70);
    height: rem(70);
    border-radius: 50%;
    overflow: hidden;
    @include for-size(desktop) {

    }
  }

  &__name {
    margin-bottom: rem(10);
    text-overflow: ellipsis;
    overflow: hidden;
  }

  &__position {
  }

  &__email {
    text-overflow: ellipsis;
    overflow: hidden;
    cursor: pointer;
  }

  .user__tooltip {
    border: none;
    -webkit-border-radius: 4px;
    -moz-border-radius: 4px;
    border-radius: 4px;
    padding: rem(3) rem(10);
    font-size: rem(14);
    color: #fefefe;
    background: #323231;
    box-shadow: rem(0) rem(3) rem(5) rgba(#323231, .1);

    &:before, &:after {
      display: none;
    }
  }
}