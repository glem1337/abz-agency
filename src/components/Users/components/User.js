import React from 'react';
import ReactTooltip from "react-tooltip";
import defaultAvatar from '../../../assets/icons/photo-cover.svg';

const User = ({user}) => (
    <div className="user">
        <img
            className="user__photo"
            src={user.photo || defaultAvatar}
            alt={user.name}
            onError={e => {
                e.target.src = defaultAvatar;
            }}
        />
        <div className="user-info-wrapper">
            <h3 className="user__name h2">{user.name}</h3>
            <p className="user__position">{user.position}</p>
            <p className="user__email" data-tip={user.email}>{user.email}</p>
            <ReactTooltip border={false} place={'bottom'} className={'user__tooltip'}/>
            <p className="user__phone">{user.phone}</p>
        </div>
    </div>

);

export default User;
