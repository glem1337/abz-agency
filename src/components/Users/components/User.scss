.user {
  padding-bottom: rem(30);
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  @include for-size(tablet) {
    margin: 0 auto;
    width: rem(200);
    padding-bottom: rem(65);
  }
  &__photo {
    border: 2px solid $light-color;
    margin: 0 auto rem(20);
    width: rem(70);
    height: rem(70);
    border-radius: 50%;
    overflow: hidden;
    @include for-size(desktop) {

    }
  }
  &__name {
    margin-bottom: rem(17);
    text-overflow: ellipsis;
    overflow: hidden;
  }
  &__position {
  }
  &__email {
    text-overflow: ellipsis;
    overflow: hidden;
    cursor: pointer;
  }
  .user__tooltip {
    border: none;
    -webkit-border-radius: 4px;
    -moz-border-radius: 4px;
    border-radius: 4px;
    padding: rem(3) rem(10);
    font-size: rem(14);
    color: #fefefe;
    background: #323231;
    box-shadow: rem(0) rem(3) rem(5) rgba(#323231, .1);
    &:before, &:after {
      display: none;
    }
  }
}