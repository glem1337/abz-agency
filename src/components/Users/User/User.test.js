import React from 'react';
import {shallow} from 'enzyme'
import User from "./User";
import {formatPhone} from "../../../utils/format/format"

describe('User', () => {

    const props = {
        user: {
            name: 'John Doe',
            photo: '/src/to/img/image.png',
            position: 'Lawyer',
            email: 'email@gmail.com',
            phone: '+380992104201',
        }
    };

    const user = shallow(<User {...props} />)

    it('renders properly', () => {
        expect(user).toMatchSnapshot();
    });

    it('should be correct data', () => {
        expect(user.find('.user__name').text()).toEqual(props.user.name);
        expect(user.find('.user__photo')).toHaveLength(1);
        expect(user.find('ReactTooltip')).toHaveLength(1);
        expect(user.find('.user__position').text()).toEqual(props.user.position);
        expect(user.find('.user__email').text()).toEqual(props.user.email);
        expect(user.find('.user__phone').text()).toEqual(formatPhone(props.user.phone));
    });


});