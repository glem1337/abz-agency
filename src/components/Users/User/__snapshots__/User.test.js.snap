// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User renders properly 1`] = `ShallowWrapper {}`;
